### DISCUSSION TITLE
Print the component stack on hydration errors

### DISCUSSION THREAD

**Discussion by remorses** [vercel/next.js#47542] [Ideas]
### Goals

1. Hydration errors are basically impossible to debug in production because there is no stack trace or anything to work with, Next should print the hydration error component stack in the console (exposed as `errorInfo.componentStack` in `onRecoverableError`)

### Background

React exposes the hydration error stack as second argument of `onRecoverableError`, `errorInfo.componentStack` (i found about it in [this tweeet](https://twitter.com/dan_abramov/status/1639800332958986240?s=20)), Next should print that stack together with the error message

Docs about onRecoverableError: https://legacy.reactjs.org/docs/react-dom-client.html#hydrateroot

Related discussion: https://github.com/vercel/next.js/discussions/36641

### Proposal

Add the component stack to the default `onRecoverableError` here

https://github.com/vercel/next.js/blob/bd8ba61bca0451cdddc313432ff50ca554025bf5/packages/next/src/client/on-recoverable-error.ts#L3

**Comment by AndersDJohnson** [comment]
Principal Engineer on the Web team at @target here - we would find this extremely useful as well!

**Comment by timneutkens** [comment]
https://twitter.com/timneutkens/status/1665715225193066496

**Comment by AndersDJohnson** [comment]
Proposing https://github.com/vercel/next.js/pull/50943 as a potential partial solution here, which also relates to https://github.com/vercel/next.js/discussions/36641#discussioncomment-5215494.
